quakemigrate/io/data.py

```diff
@@ -413,6 +398,92 @@ def __init__(self, starttime, endtime, sampling_rate, stations=None,
         self.availability = None
         self.wa_waveforms = None
 
+    def check_availability(self, st, all_channels=False, allow_gaps=False,
+                           full_timespan=True):
+        """
+        Check waveform availability against data quality criteria. There are a
+        number of hard-coded checks: for whether any data is present; for
+        whether the data is a flatline (all samples have the same value); and
+        for whether the data contains overlaps. There are a selection of
+        additional optional checks which can be specified according to the
+        onset function / user preference.
+
+        Parameters
+        ----------
+        st : `obspy.Stream` object
+            Stream containing the waveform data to check against the
+            availability criteria.
+        all_channels : bool, optional
+            Whether all supplied channels (distinguished by SEED id) need to
+            meet the availability criteria to mark the data as 'available'.
+        allow_gaps : bool, optional
+            Whether to allow gaps.
+        full_timespan : bool, optional
+            Whether to ensure the data covers the entire timespan requested;
+            note that this implicitly requires that there be no gaps.
+
+        Returns
+        -------
+        available : int
+            0 if data doesn't meet the availability requirements; 1 if it does.
+        availability : dict
+            Dict of {tr_id : available} for each unique SEED ID in the input
+            stream (available is again 0 or 1).
+
+        """
+
+        availability = {}
+
+        # Check if any channels in stream
+        if bool(st):
+            # Loop through channels with unique SEED id's
+            for tr_id in sorted([tr.id for tr in st]):
+                st_id = st.select(id=tr_id)
+                availability[tr_id] = 1
+
+                # First check data is here:
+                if not bool(st_id):
+                    availability[tr_id] = 0
+                    continue
```

Comment:
Nope, good spot. But raises something I hadn't considered - if only N is fed to this function (in the stream), then it will still report available = 1 even if `all_channels=True`. Will probably fix this by adding an `n_chans` kwarg specifying the number of channels that are expected to be there.

Reply:
(Sorted in 6fe8824)